skimage/measure/_boundary.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+
+"""
+Code copied from https://github.com/machine-shop/deepwings/blob/master/deepwings/method_features_extraction/image_processing.py#L185
```

Comment:
It seems that this code was written by one of @stefanv's students so I expect that this won't be a problem. Nevertheless you're right, we should make sure that everyone involved is in agreement and happy. @theobdt is it alright if we reuse your code here? It would probably be easiest if you could license it with a BSD-compatible license or if you explicitly stated your consent here (in case we have it). :pray: 

See [this comment on zulip](https://skimage.zulipchat.com/#narrow/stream/176555-sprint/topic/boundary.20tracing/near/203599349) by @stefanv.

Reply:
Hello @lagru, yes I personally agree with this code being reused, as long as @stefanv is also ok with it. We also worked on an initial [Cython version](https://github.com/theobdt/boundary_tracing), which might be relevant.

Reply:
Good with me!